src/core/shape_inference/include/squeeze_shape_inference.hpp

```diff
@@ -90,6 +90,18 @@ std::vector<TRShape> shape_infer(const Squeeze* op,
                              });
             }
         } else {
+            if (op->get_pytorch_dynamic_rank()) {
+                int i{};
+                if (std::any_of(arg_shape.cbegin(), arg_shape.cend(), [&](const DimType& d) {
+                        ++i;
+                        return d.is_dynamic() && d.compatible(1) && unique_axes->find(i - 1) != unique_axes->end();
```

Comment:
```suggestion
                        return d.compatible(1) && unique_axes->find(i) != unique_axes->end();
```
if `i` starts from `-1` the subtract should not be required.

Reply:
done